### PR TITLE
Migrate components to uswds, remove some unused Profile fields WIP

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Updates component usage in the Profile application pages to use `uswds` prop, mostly the alerts, modals, and additional info components.
- Some styling tweaks were needed, mostly around margin or padding and layout
- Removes personal Information fields for pronouns and sexual orientation, that were never in use in production
- I have spooled up a 'legacy branch' for the sexual orientation and pronouns work, so that they can be re-added in the future easily, that branch is here: https://github.com/department-of-veterans-affairs/vets-website/tree/Profile/aw/do-not-delete-sexual-orientation-and-pronouns-work

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/73847

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Accordions
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2024-01-25 at 1 44 09 PM | Screenshot 2024-01-25 at 1 43 52 PM |
| Desktop | Screenshot 2024-01-25 at 1 45 13 PM | Screenshot 2024-01-25 at 1 44 46 PM |

### Modals
|         | Before | After |
| ------- | ------ | ----- |
| Mobile | Screenshot 2024-01-25 at 1 42 37 PM | Screenshot 2024-01-25 at 1 42 56 PM

###  Alerts

| Before | After |
| ------ | ----- |
| Screenshot 2024-01-25 at 4 41 26 PM | Screenshot 2024-01-25 at 4 40 02 PM |
| Screenshot 2024-01-25 at 2 22 00 PM | Screenshot 2024-01-25 at 4 58 39 PM |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/personalization/common/e2eHelpers.js
Comment (allisonlu) [inline src/applications/personalization/common/e2eHelpers.js:1]: very nice

**Review by allisonlu — APPROVED** [review]
:shipit: nice work!

File: src/applications/personalization/profile/components/alerts/DirectDepositConnectionError.jsx [hunk 1]
```diff
@@ -35,7 +35,10 @@ const DirectDepositConnectionError = ({ benefitType }) => {
   }
 
   return (
-    <va-alert class="vads-u-margin-bottom--3 medium-screen:vads-u-margin-bottom--4">
+    <va-alert
+      class="vads-u-margin-bottom--3 medium-screen:vads-u-margin-bottom--4"
+      uswds
```
Comment (DanielMiller-agile6) [inline src/applications/personalization/profile/components/alerts/DirectDepositConnectionError.jsx:40]: I forgot if you mentioned already, but did you hear if there is a plan to make the `uswds` attr the default in the future? or will they support a non-uswds variant for the foreseeable future?

**Review by DanielMiller-agile6 — APPROVED** [review]
looks great! seems like these were fairly compatible with the non-uswds variety, so great news :D

File: src/applications/personalization/profile/components/alerts/DirectDepositConnectionError.jsx (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/profile/components/alerts/DirectDepositConnectionError.jsx:40]: @DanielMiller-agile6 yes the uswds prop with start to default to true in february, and so this work is to make sure that when that happens all the components have basically already been migrated/checked so we dont run into any issue.

You are right that most of the changes that have been needed from migrating to uswds versions have been minimal, so hoping the rest of this work goes as smoothly... mostly its just the chore of making sure I'm not missing anything. :)

**Pull request merged**